src/main/java/LinkedStack.java

```diff
@@ -103,5 +133,18 @@ public void setData(T data) {
         public void setNext(Node<T> next) {
             this.next = next;
         }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            Node<?> node = (Node<?>) o;
+            return Objects.equals(data, node.data);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(data, next);
+        }
```

Comment:
Chatted offline briefly, imo remove. This is needlessly wrapping `T#equals`, and also implies statements that are plainly false (lists are their nodes, after all).